File: homework4 .vb

Imports Microsoft.VisualBasic.CompilerServices
Imports System

Namespace Homework_04

    Public Class Calculator

        Public Property Height As Double
            Get
                Return Me._HeightDouble
            End Get

            Set(ByVal value As Double)
                Me._HeightDouble = value
            End Set
        End Property

        Public Property HeightUnit As String
            Get
                Return Me._HeightUnit
            End Get

            Set(ByVal value As String)
                Me._HeightUnit = value.Trim().ToLower()
            End Set
        End Property

        Public Property Weight As Double
            Get
                Return Me._WeightDouble
            End Get

            Set(ByVal value As Double)
                Me._WeightDouble = value
            End Set
        End Property

        Public Property WeightUnit As String
            Get
                Return Me._WeightUnit
            End Get

            Set(ByVal value As String)
                Me._WeightUnit = value.ToLower()
            End Set
        End Property

        Public Sub New(ByVal HeightDouble As Double, ByVal HeightUnitString As String, ByVal WeightDouble As Double, ByVal WeightUnitString As String)
            Me.Height = HeightDouble
            Me.HeightUnit = HeightUnitString
            Me.Weight = WeightDouble
            Me.WeightUnit = WeightUnitString
            Me.BMI()
        End Sub

        Public Function BMI() As Double
            Dim flag As Boolean = Operators.CompareString(Me.WeightUnit, "kg", False) = 0
            Dim num As Double
            If flag Then
                num = Me.Weight
            Else
                num = Me.Weight * 0.44999998807907104
            End If

            Dim flag2 As Boolean = Operators.CompareString(Me.HeightUnit, "m", False) = 0
            Dim x As Double
            If flag2 Then
                x = Me.Height
            Else
                x = Me.Height * 0.02500000037252903
            End If

            Return num / Math.Pow(x, 2)
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by Refactoring Essentials.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: homework4.cs
using Microsoft.VisualBasic.CompilerServices;
using System;

namespace Homework_04
{
	public class Calculator
	{
		//
		// Properties
		//
		public double Height {
			get {
				return this._HeightDouble;
			}
			set {
				this._HeightDouble = value;
			}
		}

		public string HeightUnit {
			get {
				return this._HeightUnit;
			}
			set {
				this._HeightUnit = value.Trim ().ToLower ();
			}
		}

		public double Weight {
			get {
				return this._WeightDouble;
			}
			set {
				this._WeightDouble = value;
			}
		}

		public string WeightUnit {
			get {
				return this._WeightUnit;
			}
			set {
				this._WeightUnit = value.ToLower ();
			}
		}

		//
		// Constructors
		//
		public Calculator (double HeightDouble, string HeightUnitString, double WeightDouble, string WeightUnitString)
		{
			this.Height = HeightDouble;
			this.HeightUnit = HeightUnitString;
			this.Weight = WeightDouble;
			this.WeightUnit = WeightUnitString;
			this.BMI ();
		}

		//
		// Methods
		//
		public double BMI ()
		{
			bool flag = Operators.CompareString (this.WeightUnit, "kg", false) == 0;
			double num;
			if (flag) {
				num = this.Weight;
			} else {
				num = this.Weight * 0.44999998807907104;
			}
			bool flag2 = Operators.CompareString (this.HeightUnit, "m", false) == 0;
			double x;
			if (flag2) {
				x = this.Height;
			} else {
				x = this.Height * 0.02500000037252903;
			}
			return num / Math.Pow (x, 2.0);
		}
	}
}
